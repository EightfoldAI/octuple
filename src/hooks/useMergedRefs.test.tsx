import * as React from 'react';
import { render as renderComponent } from '@testing-library/react';
import { useMergedRefs } from './useMergedRefs';

const render = (children: any): any => renderComponent(children);

it('merges object and function refs', () => {
  const refValues: any = {};
  const RefComponent: React.FC = () => {
    const refA = React.useRef<HTMLDivElement>(null);
    const refB_ = React.useRef<HTMLDivElement>(null);
    const refB = (el: HTMLDivElement): void => {
      refB_.current = el;
    };
    const ref = useMergedRefs(refA, refB);
    React.useEffect(() => {
      refValues.a = refA.current;
      refValues.b = refB_.current;
    });

    return React.createElement('div', { ref });
  };
  render(React.createElement(RefComponent));
  expect(refValues.a).toBe(refValues.b);
  expect(refValues.b).toMatchSnapshot();
});
