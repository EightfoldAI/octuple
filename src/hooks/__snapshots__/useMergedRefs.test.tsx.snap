// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merges object and function refs 1`] = `<div />`;
